File: packages/php/src/Api/BackgroundApi.php
<?php

declare(strict_types=1);

namespace AILabTools\Api;

use AILabTools\Generated\Params\CutoutUniversalBackgroundRemovalParams;
use AILabTools\Generated\Response\CutoutUniversalBackgroundRemovalResponse;

final class BackgroundApi
{
    public function __construct(private readonly CutoutApi $cutout)
    {
    }

    public function remove(
        CutoutUniversalBackgroundRemovalParams $params,
    ): CutoutUniversalBackgroundRemovalResponse {
        return $this->cutout->cutoutUniversalBackgroundRemoval($params);
    }
}

File: packages/swift/Sources/AILabTools/BackgroundAPI.swift
import Foundation

public final class BackgroundAPI {
    private let cutout: CutoutAPI

    internal init(cutout: CutoutAPI) {
        self.cutout = cutout
    }

    public func remove(
        _ parameters: CutoutUniversalBackgroundRemovalParams
    ) async throws -> CutoutUniversalBackgroundRemovalResponse {
        try await cutout.cutoutUniversalBackgroundRemoval(parameters)
    }
}
